src/robot.py

```diff
@@ -62,11 +121,60 @@ def robotInit(self):
     # Add the deploy artifacts to the shuffleboard
     addDeployArtifacts()
 
+  def disabledPeriodic(self):
+    self.drivebase.stop()
+
+    self.setOutputs()
+
   def autonomousInit(self):
     """This function is run once each time the robot enters autonomous mode."""
+    # Check only once during autonomous initialization for setting initial pose
+    self.autoSelected = self.chooser.getSelected()
+
+    match self.autoSelected:
+      # Positions mark the autonomous start line, adjust for frame and bumpers
+      case self.blueLeft:
+        kX = 76.125 - RobotConstants.frame_length / 2 - RobotConstants.bumper_width
+        kY = 275.42
+        kRotation = math.pi
+      case self.blueMiddle:
+        kX = 76.125 - RobotConstants.frame_length / 2 - RobotConstants.bumper_width
+        kY = 218.42
+        kRotation = math.pi
+      case self.blueRight:
+        kX = 76.125 - RobotConstants.frame_length / 2 - RobotConstants.bumper_width
+        kY = 161.42
+        kRotation = math.pi
+      case self.redRight:
+        kX = 577.125 + RobotConstants.frame_length / 2 + RobotConstants.bumper_width
+        kY = 275.42
+        kRotation = 0
+      case self.redMiddle:
+        kX = 577.125 + RobotConstants.frame_length / 2 + RobotConstants.bumper_width
+        kY = 218.42
+        kRotation = 0
+      case self.redLeft:
+        kX = 577.125 + RobotConstants.frame_length / 2 + RobotConstants.bumper_width
+        kY = 161.42
+        kRotation = 0
+      case _:
+        kX = 0
+        kY = 0
+        kRotation = 0
+
+    # Set the dashboard programmed initial pose.
+    inchToM = 39.73  # Inches to metres (x / inchToM)
+    if self.drivebase.isCalibrated():
+      self.drivebase.resetOdometer(
+        wpimath.geometry.Pose2d(kX / inchToM, kY / inchToM, kRotation)
+      )
+    # else the gyro is probably broken if it is still calibrating, degraded operations tbd.
 
   def autonomousPeriodic(self):
     """This function is called periodically during autonomous."""
+    self.drivebase.stop()
+
+    self.setOutputs()
```

Comment:
We tested the basic auto routine while the bot was rigged in the pit. Do we need more testing to be done before that's merged?

Reply:
Auto routine is still in a pull request for its feature branch to develop. This PR is to move the existing develop code from the practice match to main before we add changes to it.